src/niftyone/figures/generator.py

```diff
@@ -144,17 +144,14 @@ def generate(
         img = nib.nifti1.load(img_path)
         img = noimg.to_iso_ras(img)
 
-        # Handle overlays - currently only handles 1, update to handle multiple
+        # Handle overlays
         if len(records) > 1:
-            # Temporary logic to handle multiple overlays
-            if len(records) > 2:
-                raise NotImplementedError("Multi-image overlay not yet implemented")
             overlays: list[nib.Nifti1Image] = []
             for overlay_record in records[1:]:
                 overlay_path = Path(overlay_record["finfo"]["file_path"])
                 overlay = nib.nifti1.load(overlay_path)
                 overlays.append(noimg.to_iso_ras(overlay))
-                self.view_kwargs["overlay"] = overlays.pop()
+            self.view_kwargs["overlay"] = overlays
```

Comment:
Note for future, and probably have said this before, but mutating `view_kwargs` on each call, which are intended to be static, is a bit of a hack.

Reply:
After running into the figure name issue, I think rather than updating the view_kwargs this way, will just pass the overlay into `self.view_fn` as a kwarg. I don't think this is an issue at the moment, but to prevent any sort of issues down the road. Just need to make sure / update the `view_fn` calls to be able to accept the overlay kwarg.

Reply:
Overlay is directly passed into the `view_fn` as of e16c33d rather than manipulate the `self.view_kwargs`